pytensor/tensor/rewriting/subtensor_lift.py

```diff
@@ -110,108 +110,79 @@ def local_subtensor_of_dot(fgraph, node):
     return [r]
 
 
-# fast_compile to allow opt subtensor(cast{float32}(make_vector))
-@register_canonicalize("fast_compile")
+@register_canonicalize("shape_unsafe")
+@register_specialize("shape_unsafe")
```

Comment:
I don't know the difference between canonicalize and specialize.

Reply:
canonicalize happens before stabilize and before specialize. Canonicalize should simplify graphs convert different forms into a canonical one. Specialize is allowed to go bananas and produce very specialized code. Many rewrites we apply in multiple phases so they can interact with others that are specific to that phase